bigquery/google/cloud/bigquery/external_config.py

```diff
@@ -412,46 +588,99 @@ def options(self):
+        """
+        prop = self._properties.get('compression')
+        return prop
 
-    schema = _ListApiResourceProperty('schema', 'schema', SchemaField)
-    """See
-    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).schema
-    https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.schema
-    """
+    @compression.setter
+    def compression(self, value):
+        self._properties['compression'] = value
+
+    @property
+    def ignore_unknown_values(self):
+        """bool: If `True`, extra values that are not represented in the table
+        schema are ignored. Defaults to `False`.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).ignoreUnknownValues
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.ignoreUnknownValues
+        """
+        prop = self._properties.get('ignoreUnknownValues')
+        return prop
+
+    @ignore_unknown_values.setter
+    def ignore_unknown_values(self, value):
+        self._properties['ignoreUnknownValues'] = value
+
+    @property
+    def max_bad_records(self):
+        """int: The maximum number of bad records that BigQuery can ignore when
+        reading data.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).maxBadRecords
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.maxBadRecords
+        """
+        prop = self._properties.get('maxBadRecords')
+        return prop
+
+    @max_bad_records.setter
+    def max_bad_records(self, value):
+        self._properties['maxBadRecords'] = value
+
+    @property
+    def source_uris(self):
+        """List[str]: URIs that point to your data in Google Cloud.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).sourceUris
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.sourceUris
+        """
+        prop = self._properties.get('sourceUris', [])
+        return prop
+
+    @source_uris.setter
+    def source_uris(self, value):
+        self._properties['sourceUris'] = value
+
+    @property
+    def schema(self):
+        """List[google.cloud.bigquery.schema.SchemaField]: The schema for the
+        data.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).schema
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#externalDataConfiguration.schema
+        """
+        prop = self._properties.get('schema', [])
+        return prop
+
+    @schema.setter
+    def schema(self, value):
+        self._properties['schema'] = value
```

Comment:
We might want to go all the way and convert these to API format now. (Check out LoadJobConfig.schema for an example)